bayesian/utility/user_utils.py

```diff
@@ -22,42 +26,59 @@ def get_user(uuid_val):
         user = result.one()
         return user
     except SQLAlchemyError as e:
-        logger.exception(f"Error fetching user with id {uuid_val}")
+        logger.exception("Error fetching user with id %s", uuid_val)
         raise UserException("Error fetching user") from e
 
 
 @retry(reraise=True, stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_fixed(1))
-def create_user(uuid_val, user_source):
-    """Create User."""
+def create_or_update_user(user_id, snyk_api_token, user_source):
+    """Create or Update User."""
     try:
-        insert_user_stmt = insert(UserDetails).values(user_id=uuid_val, user_source=user_source,
-                                                      created_date=datetime.datetime.now())
-
-        rdb.session.execute(insert_user_stmt)
-        rdb.session.commit()
-        logger.info(f"User created with id {uuid_val}")
+        if snyk_api_token:
```

Comment:
Isn't it redundant validation? Request has been already validated by the route handler right?

Reply:
Done